### PR TITLE
MNT-24335: improve historical tasks performance (#4702)

### PR DESCRIPTION
**Pull request by vrmoreira**
(cherry picked from commit 84b6e253bc3631e45769c2750edbf939949def42)

**What kind of change does this PR introduce?**

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [X] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

## Description

- Issue Link: https://hyland.atlassian.net/browse/MNT-24335

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

### CODE REVIEW

File: activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/db/DbSqlSession.java [hunk 1]
```diff
@@ -146,6 +146,7 @@ public class DbSqlSession implements Session {
```
Comment (balsarori) [inline activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/db/DbSqlSession.java]: We'll need to add `7.11.1`
`ACTIVITI_VERSIONS.add(new ActivitiVersion("7.11.1"));`
before 
`ACTIVITI_VERSIONS.add(new ActivitiVersion(LAST_V7_VERSION));`

Also need to add the update steps

**Review by balsarori — CHANGES_REQUESTED** [review]
We'll need this commit to fix native query issue if still exists --> https://github.com/Activiti/Activiti/pull/4675/commits/ff69ebc4df9940981bbb8b6751f9f1ef68b9159c

File: activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/db/DbSqlSession.java (hunk 1 above)
Comment (vrmoreira) [reply activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/db/DbSqlSession.java]: In order to make sure that update (indexes) are run if we upgrade from a previous Activiti 8.x version, I've added schema version 8.1.1 and add the steps. This won't stop to re-execute the addition of those indexes if we upgrade from the latest Activiti 7.11.x version.

I'll add the steps for 7.11.1 version but need to verify if there's any problem executing those steps twice.

Comment (vrmoreira) [reply activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/db/DbSqlSession.java]: Found a couple of problems, which think I've nailed it.
Also, on "testcontainers" branch, I've updated the `DbSchemaUpdateTest` in order to test it against all databases. I couldn't see any problems but, locally, on certain databases, if I change the updatestep for 8.1.1 schema by adding an error, everything runs ok, which is weird.

File: activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/cfg/ProcessEngineConfigurationImpl.java [hunk 2]
```diff
@@ -1146,18 +1147,31 @@ public void initDatabaseType() {
...
+      databaseType = databaseVendor;
+
+      switch (databaseVendor) {
+          case DATABASE_TYPE_MYSQL:
+              String databaseProductVersion = databaseMetaData.getDatabaseProductVersion();
+              // MariaDB has performance penalty when using "withoutJoins" SQL scripts
+              // example of MariaDB 10.5.4
+              //     input: vendor=MySQL, version=5.5.5-10.5.24-MariaDB-1:10.5.24+maria~ubu2004
+              //     output: databaseType=mariadb
+              if (databaseProductVersion.contains("MariaDB")) {
```
Comment (balsarori) [inline activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/cfg/ProcessEngineConfigurationImpl.java]: maybe we should do a case insensitive comparison?

File: activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/cfg/ProcessEngineConfigurationImpl.java [hunk 3]
```diff
@@ -1146,18 +1147,31 @@ public void initDatabaseType() {
...
+      databaseType = databaseVendor;
+
+      switch (databaseVendor) {
+          case DATABASE_TYPE_MYSQL:
+              String databaseProductVersion = databaseMetaData.getDatabaseProductVersion();
+              // MariaDB has performance penalty when using "withoutJoins" SQL scripts
+              // example of MariaDB 10.5.4
+              //     input: vendor=MySQL, version=5.5.5-10.5.24-MariaDB-1:10.5.24+maria~ubu2004
+              //     output: databaseType=mariadb
+              if (databaseProductVersion.toLowerCase().contains("mariadb")) {
```
Comment (balsarori) [inline activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/cfg/ProcessEngineConfigurationImpl.java]:
```suggestion
              if (databaseProductVersion.toLowerCase().contains(DATABASE_TYPE_MARIADB)) {
```

**Review by balsarori — APPROVED** [review]

**Pull request merged**